### ISSUE TITLE
"Implement interface" TS refactoring adds extra semicolon in type

### ISSUE THREAD

**Issue by joyceerhl** [microsoft/TypeScript#50244]
Does this issue occur when all extensions are disabled?: Yes

- VS Code Version: 1.71-insider
- OS Version: Windows 11

Steps to Reproduce:

1. Have the following code in the microsoft/vscode repo
```ts
interface IFoo {
	someFn(): { [key: string]: string | null };
}

class Foo implements IFoo {
	
}
```
2. Select to implement interface IFoo 
![image](https://user-images.githubusercontent.com/30305945/183766721-ff0927cd-06d9-4521-93b2-c53c7cdaefbf.png)
3. :bug: generated implementation violates our eslint rules
<img width="928" alt="image" src="https://user-images.githubusercontent.com/30305945/183766961-008769de-646e-41ad-83e5-c05ee4e96d31.png">

**Comment by a-tarasyuk** [comment]
@mjbvz  As I know - `;` is a valid delimiter for members of TypeLiteral. [Example](https://www.typescriptlang.org/play?ts=4.8.0-beta#code/JYOwLgpgTgZghgYwgAgJIDED2nkG8BQAkAM6YC2E6IAFAJQBceyA2gNYQCejxYUoA5gF1uvAcgA+yEAFcANrOQBfANz5F+fAllxixZFhzAyAB1kQK4PRmx58ye8lIUqdRrhbsuj0SCEi+vhJScrLKSrYOkchgABZQmADuUhBJAKJQ8VDUAEQAshCxmAAmUphgyEam5hDgEEUAdNm0qpHq6kA).

/cc @RyanCavanaugh

**Comment by mjbvz** [comment]
@a-tarasyuk Yes the code is valid TypeScript but in our case the extra trailing `;` causes eslint errors. Since this `;` could be omitted here, the feature request is to only add `;` if it is really needed

**Comment by RyanCavanaugh** [comment]
There are also lint rules which require trailing terminators, of course.

It seems like if your lint configuration is going to be this picky, having fix-on-save would be a requirement anyway.

*Maybe* a sensible rule would be to re-use the original type annotation's text *if* it contains no newlines?
